int32 center_x
int32 center_y
int32 centroid_x
int32 centroid_y
time time
